sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ManagementChannel.java

```diff
@@ -151,6 +202,113 @@
+                                 Class<T> responseType) {
+
+        messageToSchedule.setScheduledEnqueueTime(scheduledEnqueueTime);
+
+        return cbsAuthorizationOnce()
+            .then(
+                channelMono.flatMap(requestResponseChannel -> {
+                    List<Message> messagesToSchedule = new ArrayList<>();
+                    messagesToSchedule.add(messageSerializer.serialize(messageToSchedule));
+                    Map<String, Object> requestBodyMap = new HashMap<>();
+                    Collection<HashMap<String, Object>> messageList = new LinkedList<>();
+                    for (Message message : messagesToSchedule) {
+                        HashMap<String, Object> messageEntry = new HashMap<>();
+                        Pair<byte[], Integer> encodedPair;
+                        encodedPair = MessageUtils.encodeMessageToOptimalSizeArray(message,
+                            MAX_MESSAGE_LENGTH_SENDER_LINK_BYTES);
+
+                        messageEntry.put(REQUEST_RESPONSE_MESSAGE, new Binary(encodedPair.getFirstItem(),
+                            0, encodedPair.getSecondItem()));
+                        messageEntry.put(REQUEST_RESPONSE_MESSAGE_ID, message.getMessageId());
+                        messageList.add(messageEntry);
+                    }
+                    requestBodyMap.put(REQUEST_RESPONSE_MESSAGES, messageList);
+
+                    Message requestMessage = createRequestMessageFromValueBody(
+                        REQUEST_RESPONSE_SCHEDULE_MESSAGE_OPERATION, requestBodyMap,
+                        MessageUtils.adjustServerTimeout(DEFAULT_REQUEST_RESPONSE_TIMEOUT));
+                    return requestResponseChannel.sendWithAck(requestMessage);
+                })
+            )
+             .map(responseMessage -> {
+
+                    int statusCode = RequestResponseUtils.getResponseStatusCode(responseMessage);
+                    List<T> sequenceNumbers = null;
+                    if (statusCode ==  REQUEST_RESPONSE_OK_STATUS_CODE) {
+                        @SuppressWarnings("unchecked")
+                        Map<String, Object> responseBody = (Map<String, Object>) ((AmqpValue) responseMessage
+                            .getBody()).getValue();
+                        Object seqNumberListObj = responseBody.get(REQUEST_RESPONSE_SEQUENCE_NUMBERS);
+
+                        if (seqNumberListObj instanceof  long[]){
+                            @SuppressWarnings("unchecked")
+                            List<T> seqNumbers  = (List<T>)Arrays.stream((long[])seqNumberListObj)
+                                .boxed().collect(Collectors.toList());
+                            sequenceNumbers = seqNumbers;
+                        }
+                    }
+                    return sequenceNumbers;
+                })
+         .flatMapMany(Flux::fromIterable);
+    }
+
+    private Message createRequestMessageFromValueBody(String operation, Object valueBody, Duration timeout) {
```

Comment:
I don't like the use of Object. I'd prefer it be a strongly typed Map or something similar.